### PR TITLE
use twig in Structure and Variables controller and add missing tests #14801

### PR DESCRIPTION
**Pull request by seyfer**
Signed-off-by: Oleg Abrazhaev <seyferseed@gmail.com>

### Description

* use twig templates in two controllers
* add missing tests
* minor fixes btw

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]

**Review by ibennetch — APPROVED** [review]
Looks great to me, thank you.

Requested additional review from Maurício, our Twig expert.

File: templates/database/structure/overhead.twig [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<a href="{{ url }}" id="overhead">
```
Comment (mauriciofauth) [inline templates/database/structure/overhead.twig]: I think it's better to call the function inside the template:
```suggestion
<a href="{{ url('/table/structure', table_url_params) }}#showusage" id="overhead">
```

File: libraries/classes/Controllers/Database/StructureController.php [hunk 2]
```diff
@@ -393,20 +398,21 @@ protected function displayTableList(): string
...
             if ($this->isShowStats) {
                 $overhead = '-';
                 if ($formatted_overhead != '') {
-                    $overhead = '<a href="' . Url::getFromRoute('/table/structure', $tableUrlParams)
-                        . '#showusage">'
-                        . '<span>' . $formatted_overhead . '</span>&nbsp;'
-                        . '<span class="unit">' . $overhead_unit . '</span>'
-                        . '</a>' . "\n";
+                    $overhead = $this->template->render('database/structure/overhead', [
+                        'url' => Url::getFromRoute('/table/structure', $tableUrlParams) . '#showusage',
```
Comment (mauriciofauth) [inline libraries/classes/Controllers/Database/StructureController.php]: I think it's better to call the function inside the template:
```suggestion
                        'table_url_params' => $tableUrlParams,
```

File: libraries/classes/Controllers/Server/VariablesController.php [hunk 3]
```diff
@@ -215,12 +215,17 @@ private function formatVariable($name, $value)
...
-                        . htmlspecialchars(
-                            implode(' ', Util::formatByteDown($value, 3, 3))
+                    $formattedValue = trim(
+                        $this->template->render(
+                            'server/variables/format_variable',
+                            [
+                                'valueTitle' => htmlspecialchars(Util::formatNumber($value, 0)),
+                                'value' => htmlspecialchars(
+                                    implode(' ', Util::formatByteDown($value, 3, 3))
+                                ),
```
Comment (mauriciofauth) [inline libraries/classes/Controllers/Server/VariablesController.php]: You don't need to escape the variables with `htmlspecialchars` here.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

File: libraries/classes/Controllers/Server/VariablesController.php (hunk 3 above)
Comment (seyfer) [reply libraries/classes/Controllers/Server/VariablesController.php]: @mauriciofauth it was in the code before, so I kept it. Shall I remove it?

Comment (seyfer) [reply libraries/classes/Controllers/Server/VariablesController.php]: ah true, Twig does it :+1:

**Review by MauricioFauth — APPROVED** [review]

**Comment by mauriciofauth** [comment]
Merged, thanks for your contribution!

**Pull request merged**